src/Uno.UI/Microsoft/UI/Xaml/Controls/ColorPicker/ColorHelpers.cs

```diff
@@ -486,27 +555,23 @@ int sgn(T val)
 			return bitmap;
 		}
 
-		// UNO TODO: This method is only used when drawing the color spectrum using WinUI composition API's which are currently unsupported.
-		// This method is only partially ported to C#.
-		/*public static LoadedImageSurface CreateSurfaceFromPixelData(
+		public static LoadedImageSurface CreateSurfaceFromPixelData(
```

Comment:
Keep in mind the ColorPicker has two spectrum rendering paths:

 1. As writable bitmaps normally
 2. Using a special composition brush for performance on supported systems

Since I highly doubt Uno now supports composition brushes and the composition system for rendering the spectrum, you probably should keep some comments here. Porting this method over to C# more fully is also probably fine for future-proofing the code but again without composition brushes is unnecessary.

Reply:
You're right. I stopped half-way through and decided it isn't worth it, but I kept the parts that I did port so that whoever needs to port the code later doesn't have to duplicate the work.

Reply:
Can you add some of the removed comments back then?

```
// UNO TODO: This method is only used when drawing the color spectrum using WinUI composition API's which are currently unsupported.
// This method is only partially ported to C#.
```